receiver/awscontainerinsightreceiver/internal/k8swindows/kubelet.go

```diff
@@ -74,17 +74,12 @@ func (k *kubeletSummaryProvider) getPodMetrics(summary *stats.Summary) ([]*cExtr
 		tags[ci.K8sNamespace] = pod.PodRef.Namespace
 		tags[ci.Timestamp] = strconv.FormatInt(pod.CPU.Time.UnixNano(), 10)
 
-		// CPU metric
-		metric.AddField(ci.MetricName(ci.TypePod, ci.CPUTotal), float64(*pod.CPU.UsageCoreNanoSeconds))
-		metric.AddField(ci.MetricName(ci.TypePod, ci.CPUUtilization), float64(*pod.CPU.UsageCoreNanoSeconds)/float64(nodeCPUCores))
-
-		// Memory metrics
-		metric.AddField(ci.MetricName(ci.TypePod, ci.MemUsage), *pod.Memory.UsageBytes)
-		metric.AddField(ci.MetricName(ci.TypePod, ci.MemRss), *pod.Memory.RSSBytes)
-		metric.AddField(ci.MetricName(ci.TypePod, ci.MemWorkingset), *pod.Memory.WorkingSetBytes)
-		metric.AddField(ci.MetricName(ci.TypePod, ci.MemReservedCapacity), k.hostInfo.GetMemoryCapacity())
-		metric.AddField(ci.MetricName(ci.TypePod, ci.MemUtilization), float64(*pod.Memory.WorkingSetBytes)/float64(k.hostInfo.GetMemoryCapacity())*100)
-
+		rawMetric := extractors.ConvertPodToRaw(&pod)
+		for _, extractor := range GetMetricsExtractors() {
+			if extractor.HasValue(rawMetric) {
```

Comment:
I don't love the pattern of having to check with `HasMetric` before calling `GetValue`.  That seems like something that could be done in one method.  

That being said, I checked the existing code and I see it is an established pattern so I am 100% fine with continuing the pattern.  A refactor or breaking the pattern is not worth while